webapp/src/Controller/Team/SubmissionController.php

```diff
@@ -60,47 +61,111 @@ public function createAction(Request $request, ?Problem $problem = null): Respon
             if ($contest === null) {
                 $this->addFlash('danger', 'No active contest');
             } elseif (!$this->dj->checkrole('jury') && !$contest->getFreezeData()->started()) {
                 $this->addFlash('danger', 'Contest has not yet started');
             } else {
-                /** @var Problem $problem */
-                $problem = $form->get('problem')->getData();
-                /** @var Language $language */
-                $language = $form->get('language')->getData();
-                /** @var UploadedFile[] $files */
-                $files      = $form->get('code')->getData();
-                if (!is_array($files)) {
-                    $files = [$files];
+                $problem = null;
+                $language = null;
+                $files = [];
+                $entryPoint = null;
+                $message = '';
+        
+                if ($formUpload->isSubmitted() && $formUpload->isValid()) {
+                    $problem = $formUpload->get('problem')->getData();
+                    $language = $formUpload->get('language')->getData();
+                    $files = $formUpload->get('code')->getData();
+                    if (!is_array($files)) {
+                        $files = [$files];
+                    }
+                    $entryPoint = $formUpload->get('entry_point')->getData() ?: null;
+                } elseif ($formPaste->isSubmitted() && $formPaste->isValid()) {
+                    $problem = $formPaste->get('problem')->getData();
+                    $language = $formPaste->get('language')->getData();
+                    $codeContent = $formPaste->get('code_content')->getData();
+        
+                    if ($codeContent == null || empty(trim($codeContent))) {
+                        $this->addFlash('danger', 'No code content provided.');
+                        return $this->redirectToRoute('team_index');
+                    }
+        
+                    $tempDir = sys_get_temp_dir();
+                    $tempFileName = sprintf(
+                        'submission_%s_%s_%s.%s',
+                        $user->getUsername(),
+                        $problem->getName(),
+                        date('Y-m-d_H-i-s'),
+                        $language->getExtensions()[0]
+                    );
+                    $tempFileName = preg_replace('/[^a-zA-Z0-9_.-]/', '_', $tempFileName);
+                    $tempFilePath = $tempDir . DIRECTORY_SEPARATOR . $tempFileName;
+                    file_put_contents($tempFilePath, $codeContent);
+        
+                    $uploadedFile = new UploadedFile(
+                        $tempFilePath,
+                        $tempFileName,
+                        'application/octet-stream',
+                        null,
+                        true
+                    );
+        
+                    $files = [$uploadedFile];
+                    $entryPoint = $tempFileName;
```

Comment:
The filename as entry point does not work for Kotlin, nor for Java.

See https://github.com/DOMjudge/domjudge/blob/main/webapp/templates/team/partials/submit_scripts.html.twig#L3 or https://github.com/DOMjudge/domjudge/blob/main/submit/submit#L206

Reply:
Have you considered ignoring the entry in Paste mode, since only one file can be uploaded with Paste?

Reply:
One example is Java, where the entry point basically specifies the class. If you would make the name of the file clear in the UI (or even an input that can be changed after selecting the problem) we could do the same auto-detection (with possibility to change it) with the entry point as for the uploaded file.

Reply:
I think it's feasible in terms of implementation, but Paste code is essentially designed for convenience with a single file. If we allow changes to the file name, it somewhat defeats the original purpose of this feature. Alternatively, we could try handling languages where the entry point is not the file name with special treatment? (Actually, this seems similar to the original approach for uploads.)

Reply:
By the way, I'm currently teaching a data structures class where only C language is allowed, and the usage of Paste is significantly higher than that of upload.

Reply:
Assume I am coding Java in my IDE and the IDE automatically adds a `package foo;` statement on top of my file with the class `Bar` and `Goo` in it. Now I **need** to be able to configure the  entry point after copy/pasting to say `foo.Bar` or `foo.Goo` (depending where the main method is defined, or if in both which one I want to be calling).

Reply:
Alright, this is a bit difficult for me. I don't know Java and Kotlin. I think I need to learn more about these two languages.

Reply:
I made a modification so that when the language is Java or Kotlin, the entry_point is always required, regardless of whether the language itself is set to require an entry_point. For other languages, the entry_point will still default to the filename as originally designed. I've tested it myself and it seems to work fine. Is this modification acceptable?

Reply:
I don't think this is the desired approach. Better would be to mimic our javascript logic: https://github.com/DOMjudge/domjudge/blob/5153f91192dcac133985cb36d7c8689387c7ac31/webapp/templates/team/partials/submit_scripts.html.twig#L42-L52

This still means if you copy something with a package it won't work, but that issue exists also in the current code with autodetection.

Reply:
> Additionally, I'd like to ask how I can reference code in a comment.

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet#linking-to-code